modules/contacts/modules.php

```diff
@@ -396,16 +420,42 @@ protected function output() {
                         $this->html_safe($c->value('phone_number')).'</a></td>'.
                         '<td class="text-end" style="width : 100px">';
                     if (in_array($c->value('type').':'.$c->value('source'), $editable, true)) {
-                        $res .= '<a data-id="'.$this->html_safe($c->value('id')).'" data-type="'.$this->html_safe($c->value('type')).'" data-source="'.$this->html_safe($c->value('source')).
-                            '" class="delete_contact cursor-pointer" title="'.$this->trans('Delete').'"><i class="bi bi-trash3 text-danger ms-2"></i></a>'.
-                            '<a href="?page=contacts&amp;contact_id='.$this->html_safe($c->value('id')).'&amp;contact_source='.
+                        $delete_attrs = 'data-id="'.$this->html_safe($c->value('id')).'" data-type="'.$this->html_safe($c->value('type')).'" data-source="'.$this->html_safe($c->value('source')).'"';
+                        if ($c->value('type') == 'ldap' && $c->value('all_fields')) {
+                            $all_fields = $c->value('all_fields');
+                            if (isset($all_fields['dn'])) {
+                                $delete_attrs .= ' data-ldap-dn="'.$this->html_safe($all_fields['dn']).'"';
+                            }
+                        }
+                        $edit_url = '?page=contacts&amp;contact_id='.$this->html_safe($c->value('id')).'&amp;contact_source='.
                             $this->html_safe($c->value('source')).'&amp;contact_type='.
-                            $this->html_safe($c->value('type')).'&amp;contact_page='.$current_page.
-                            '" class="edit_contact cursor-pointer" title="'.$this->trans('Edit').'"><i class="bi bi-gear ms-2"></i></a>';
+                            $this->html_safe($c->value('type')).'&amp;contact_page='.$current_page;
+                        
+                        // Add DN parameter for LDAP contacts lookup
+                        if ($c->value('type') == 'ldap' && $c->value('all_fields')) {
```

Comment:
There is too much ldap exposure here - try to put this in ldap_contacts module and use a specific ldap contact type subclass of Hm_Contact where all this manipulation of dn and other fields will happen.